packages/app/src/cli/models/app/loader.ts

```diff
@@ -400,6 +400,22 @@ class AppLoader {
 
     const extensions = await Promise.all(extensionPromises)
     const allExtensions = getArrayRejectingUndefined(extensions.flat())
+
+    // Validate that all extensions have a unique handle.
+    allExtensions.forEach((extension) => {
+      const matchingExtensions = allExtensions.filter((ext) => ext.handle && ext.handle === extension.handle)
+      if (matchingExtensions.length > 1) {
+        const result = joinWithAnd(matchingExtensions.map((ext) => ext.configuration.name))
+        const handle = outputToken.yellow(extension.handle)
```

Comment:
UI-wise, usually we use `cyan` for user input. Not a big deal, but maybe something to consider for consistency.

Also the whole function performs a scan of all the extensions every time. Could we make this a bit more efficient by doing something like:

```js
let handles = new Set();
allExtensions.forEach((extension) => {
  if (extension.handle && handles.has(extension.handle)) {
    const matchingExtensions = allExtensions.filter((ext) => ext.handle === extension.handle)
    const result = joinWithAnd(matchingExtensions.map((ext) => ext.configuration.name))
    const handle = outputToken.yellow(extension.handle)

    this.abortOrReport(
      outputContent`Duplicated handle "${handle}" in extensions ${result}. Handle needs to be unique per extension.`,
      undefined,
      extension.configurationPath,
    )
  } else if (extension.handle) {
    handles.add(extension.handle);
  }
});
```